google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,522 @@ def run(
+            encryption_spec_key_name (str):
+                Optional. Customer-managed encryption key spec for a
+                ModelDeploymentMonitoringJob. If set, this
+                ModelDeploymentMonitoringJob and all
+                sub-resources of this
+                ModelDeploymentMonitoringJob will be secured by
+                this key.
+
+            create_request_timeout (int):
+                Optional. Timeout in seconds for the model monitoring job creation request
+
+        Returns:
+            An instance of ModelDeploymentMonitoringJob
+        """
+        if not display_name:
+            display_name = cls._generate_display_name()
+
+        utils.validate_display_name(display_name)
+
+        if labels:
+            utils.validate_labels(labels)
+
+        if stats_anomalies_base_directory:
+            stats_anomalies_base_directory = gca_io_compat.GcsDestination(
+                output_uri_prefix=stats_anomalies_base_directory
+            )
+
+        if encryption_spec_key_name:
+            encryption_spec_key_name = gca_encryption_spec_compat.EncryptionSpec(
+                kms_key_name=encryption_spec_key_name
+            )
+
+        self = cls._empty_constructor(
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        parent = initializer.global_config.common_location_path(
+            project=self.project,
+            location=self.location,
+        )
+
+        api_client = self.api_client
+
+        _LOGGER.log_create_with_lro(cls)
+
+        if isinstance(endpoint, str):
+            endpoint = aiplatform.Endpoint(endpoint).resource_name
+        else:
+            endpoint = endpoint.resource_name
+
+        mdm_objective_config_seq = cls._parse_configs(
+            objective_configs,
+            endpoint,
+            deployed_model_ids,
+        )
+
+        gapic_mdm_job = (
+            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
+                display_name=display_name,
+                endpoint=endpoint,
+                model_deployment_monitoring_objective_configs=mdm_objective_config_seq,
+                logging_sampling_strategy=logging_sampling_strategy.as_proto(),
+                model_deployment_monitoring_schedule_config=schedule_config.as_proto(),
+                model_monitoring_alert_config=alert_config.as_proto(),
+                predict_instance_schema_uri=predict_instance_schema_uri,
+                analysis_instance_schema_uri=analysis_instance_schema_uri,
+                sample_predict_instance=sample_predict_instance,
+                stats_anomalies_base_directory=stats_anomalies_base_directory,
+                enable_monitoring_pipeline_logs=enable_monitoring_pipeline_logs,
+                labels=labels,
+                encryption_spec=encryption_spec_key_name,
+            )
+        )
+
+        gca_mdm_job = api_client.create_model_deployment_monitoring_job(
```

Comment:
This would be simpler to assign `self._gca_resource` directly here.

Reply:
fixed